Product Collection - Cross-sell product: fix Product Selector when the selected product is deleted

gigitux opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

The `ProductPicker` component was re-rendering due to the use of `useBlockProps`. Attributes are set up long the entire the Product Collection components tree and this causes the re-rendering. Since `useBlockProps` isn't necessary in this context, it has been removed to improve performance and stability.


Also, fixing the `tests/editor/create-woocommerce-blocks.spec.js` E2E, I noticed duplicate `useBlockProps` usage across different block.

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes #54345 (WOOPLUG-2976).



### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

| Before | After |
| ------ | ----- |
| <video src="https://github.com/user-attachments/assets/a59c26f5-c9f2-477b-8cb6-9182b56b46ee" />   |  <video src="https://github.com/user-attachments/assets/4a64db48-a74c-4d0c-9280-22bfb3efbaf5" />     |






<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create two products
2. Go to Pages > Add New Page
3. Add a Product Collection block and select the `Cross-Sells` (or `Upsells`) option.
4. Select one of the products you just created and save.
5. Delete the product permanently and refresh the page you saved in step 4
6. You will see a form to pick a replacement like this

![Image](https://github.com/user-attachments/assets/43fe4332-f7e1-4a9e-8a13-feb6b3cbd09e)

7. Pick a product on the list.
8. Verify that you need to select the product only once.


## Reviews By Category
1. Create a new page.
2. Add the `reviews by category` block.
3. Ensure that the block wrapper attributes aren't duplicated in two nested divs:

| Before | After |
|--------|--------|
| <img width="1796" alt="image" src="https://github.com/user-attachments/assets/ed05176e-c20f-4254-a5b6-9fc7aa02acda" /> | <img width="1822" alt="image" src="https://github.com/user-attachments/assets/2a4aed80-a5db-4370-aece-8f444d61fb72" /> | 


### Product Collection
1. Create a new page.
2. Add the `Product Collection` block.
3. Select a collection
4. Ensure that the block wrapper attributes aren't duplicated in two nested divs:

| Before | After |
|--------|--------|
| <img width="1843" alt="image" src="https://github.com/user-attachments/assets/f0f02d0e-51a5-4e89-a50d-177a815a5d1d" /> | <img width="1799" alt="image" src="https://github.com/user-attachments/assets/84b69702-4979-40ab-8181-5a0a923b5935" /> |
 


<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->
Product Collection - Cross-sell product: fix Product Selector when the selected product is deleted.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

gigitux committed:
Refactor ProductPicker component avoid not necessary re-rendering removing useBlockProps

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into fix/wooplug-2976-product-collection-cross-sells-product-selection-error-when

gigitux committed:
fix lint errors

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

gigitux committed:
Merge branch 'fix/wooplug-2976-product-collection-cross-sells-product-selection-error-when' of github.com:woocommerce/woocommerce into fix/wooplug-2976-product-collection-cross-sells-product-selection-error-when

gigitux commented on plugins/woocommerce/client/blocks/assets/js/blocks/product-collection/edit/ProductPicker.tsx:
```diff
@@ -2,16 +2,13 @@
 import { createInterpolateElement } from '@wordpress/element';
 import {
 	Placeholder,
-	// @ts-expect-error Using experimental features
```
Removed this comment because after updating to `@wordpress/components` is not necessary anymore.

gigitux commented:
Added as reviewer @kmanijak given that could have more context.

kmanijak reviewed (approved):
Huh, good catch @gigitux!

Tested Upsells and Cross-Sells, product picker picks up at first choice!

gigitux committed:
Merge branch 'trunk' into fix/wooplug-2976-product-collection-cross-sells-product-selection-error-when

gigitux committed:
Merge branch 'trunk' into fix/wooplug-2976-product-collection-cross-sells-product-selection-error-when

gigitux committed:
fix E2E test

gigitux committed:
Merge branch 'trunk' into fix/wooplug-2976-product-collection-cross-sells-product-selection-error-when

gigitux committed:
Merge branch 'fix/wooplug-2976-product-collection-cross-sells-product-selection-error-when' of github.com:woocommerce/woocommerce into fix/wooplug-2976-product-collection-cross-sells-product-selection-error-when

gigitux committed:
fix lint error

gigitux committed:
fix E2E test

gigitux committed:
Merge branch 'trunk' into fix/wooplug-2976-product-collection-cross-sells-product-selection-error-when

gigitux commented:
@kmanijak I didn't realize that the automerge was enabled, but fixing the `tests/editor/create-woocommerce-blocks.spec.js ` E2E test, I noticed duplicate `useBlockProps` usage across different blocks, so I fixed them: I updated the testing instruction. Can you review my changes? I will fix any comment in a follow-up PR! Thanks! 🙇

kmanijak commented on plugins/woocommerce/client/blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx:
```diff
@@ -155,7 +155,7 @@ const ProductCollectionContent = ( {
 	}
 
 	return (
-		<div { ...blockProps }>
```
@gigitux  this one is needed. After choosing a collection (initial prompt), this is the main wrapper of the Product Collection. Without it Product Collection block is not highlighted in editor 

https://github.com/user-attachments/assets/3b61cef0-7f7d-4e44-a5a0-3e2ac7747a1c

kmanijak commented on plugins/woocommerce/client/blocks/assets/js/blocks/product-collection/edit/ProductPicker.tsx:
```diff
@@ -52,36 +48,34 @@ const ProductPicker = (
 		  );
 
 	return (
-		<div { ...blockProps }>
```
This one as well. Basically in the  https://github.com/woocommerce/woocommerce/blob/f861f01d3c9cc1c0d632bbb4e651f829654c41d7/plugins/woocommerce/client/blocks/assets/js/blocks/product-collection/edit/index.tsx you can see there's a choice which component is rendered and any of them can be a "main" block and each of them should get block props.

kmanijak reviewed (commented):
> @kmanijak I didn't realize that the automerge was enabled, but fixing the tests/editor/create-woocommerce-blocks.spec.js  E2E test, I noticed duplicate useBlockProps usage across different blocks, so I fixed them: I updated the testing instruction. Can you review my changes? I will fix any comment in a follow-up PR! Thanks! 🙇

They're actually needed, I left two comments. Let me know when you plan to address it. I can spin up a quick PR, would be good to bring it back before Monday's code freeze.
